Lib/typing.py

```diff
@@ -100,70 +102,199 @@
 # legitimate imports of those modules.
 
 
-def _qualname(x):
-    if sys.version_info[:2] >= (3, 3):
-        return x.__qualname__
-    else:
-        # Fall back to just name.
-        return x.__name__
+def _type_check(arg, msg):
+    """Check that the argument is a type, and return it (internal helper).
 
+    As a special case, accept None and return type(None) instead.
+    Also, _TypeAlias instances (e.g. Match, Pattern) are acceptable.
+    The msg argument is a human-readable error message, e.g::
 
-def _trim_name(nm):
-    whitelist = ('_TypeAlias', '_ForwardRef', '_TypingBase', '_FinalTypingBase')
-    if nm.startswith('_') and nm not in whitelist:
-        nm = nm[1:]
-    return nm
+        "Union[arg, ...]: arg should be a type."
 
+    We append the repr() of the actual value (truncated to 100 chars).
+    """
+    if arg is None:
+        return type(None)
+    if isinstance(arg, str):
+        return ForwardRef(arg)
+    if (
+        isinstance(arg, _GenericAlias) and
+        arg.__origin__ in (Generic, _Protocol, ClassVar) or
+        arg in (Generic, _Protocol)
+        or isinstance(arg, _SpecialForm) and arg != Any
```

Comment:
I replaced ``!=`` with ``is not``, it makes sense since ``Any`` is a singleton. (If you are asking about the multiple lines within ``(...)``, then this is how it is historically preferred in typing/mypy/etc.)

Reply:
I were wondering if PEP 8 specifies whether to insert a line break. In this expression line breaks are inserted before and after boolean binary operators.

Actually PEP 8 [specifies this](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator), but contradicts to itself. It contains [examples](https://www.python.org/dev/peps/pep-0008/#indentation) with line breaks before and after boolean binary operators.

Reply:
Hm, previously we used breaks after binary ops. I will stick to this rule.